test/common/http/ip_detection_extensions.cc

```diff
@@ -0,0 +1,25 @@
+#include "ip_detection_extensions.h"
+
+#include "extensions/http/original_ip_detection/custom_header/custom_header.h"
+#include "extensions/http/original_ip_detection/xff/xff.h"
+
+namespace Envoy {
+
+Http::OriginalIPDetectionSharedPtr getXFFExtension(uint32_t hops) {
+  return std::make_shared<Extensions::Http::OriginalIPDetection::Xff::XffIPDetection>(hops);
+}
+
+Http::OriginalIPDetectionSharedPtr getCustomHeaderExtension(const std::string& header_name) {
```

Comment:
@rgs1 I think this file violates that extensions should be independent of each other, requiring both extensions to be enabled.

WDYT is the right way to get around this? Does this just null the point of this file?

Reply:
This file is there to workaround the `Envoy::Http` vs `Envoy::Extensions::Http` namespaces pollution. How why are both extensions required?

Reply:
I think from the bazel perspective the if you disable one of them here: https://github.com/envoyproxy/envoy/blob/19abe1712849fc9c807a25609f36caa07442a2ff/source/extensions/extensions_build_config.bzl#L274

it'll only stop the factory registration. So if you want to test just one of them, and you use this file, both of these extensions end up being linked into the binary I believe with the other one (or N other IP detection extension if this pattern is repeated in the future)  would just end up as dead code. 

Seems like it's probably better to link the minimal set of what you'd use in a given situation.

Reply:
possibly related - i was trying to update the envoy checkout on envoy-mobile (https://github.com/envoyproxy/envoy-mobile/pull/1481)

i had to link in the xff code - its failing, possibly for some other reason - but something tells me that this will also cause an issue

not totally sure - as xff at least is listed in `all_extensions.bzl` which kinda overrides `envoy_build_config.bzl`

Reply:
hmm it's also here:

```
 # The map may be overridden by extensions specified in envoy_build_config.
 _required_extensions = {
     "envoy.common.crypto.utility_lib": "//source/extensions/common/crypto:utility_lib",
+    "envoy.http.original_ip_detection.xff": "//source/extensions/http/original_ip_detection/xff:config",
     "envoy.request_id.uuid": "//source/extensions/request_id/uuid:config",
     "envoy.transport_sockets.tls": "//source/extensions/transport_sockets/tls:config",
 }
 ```